jsapp/js/components/common/SimpleTable.tsx

```diff
@@ -0,0 +1,36 @@
+import {Table, type TableData} from '@mantine/core';
+
+interface SimpleTableProps {
+  head: TableData['head'];
+  body: TableData['body'];
+  /**
+   * Passing minimum width enables contextual horizontal scrollbar (i.e. without
+   * it the table will never display scrollbar - regardless of how small
+   * the screen is).
+   */
+  minWidth?: number;
+}
+
+/**
+ * A wrapper component for `Table` from `@mantine/core`. It requires column
+ * headings, column data, and has optional minimum width.
+ */
+export default function SimpleTable(props: SimpleTableProps) {
+  const table = (
+    <Table
+      data={{head: props.head, body: props.body}}
+      horizontalSpacing='sm'
+      verticalSpacing='sm'
+    />
+  );
+
+  if (props.minWidth) {
+    return (
+      <Table.ScrollContainer minWidth={props.minWidth} type='native'>
+        {table}
+      </Table.ScrollContainer>
+    );
+  }
+
+  return table;
+}
```

Comment:
import {Table} from '@mantine/core';
import type {MantineStyleProps, TableData} from '@mantine/core';

interface SimpleTableProps extends MantineStyleProps {
  head: TableData['head'];
  body: TableData['body'];
  /**
   * Passing minimum width enables contextual horizontal scrollbar (i.e. without
   * it the table will never display scrollbar - regardless of how small
   * the screen is).
   */
  minWidth?: number;
}

/**
 * A wrapper component for `Table` from `@mantine/core`. It requires column
 * headings, column data, and has optional minimum width.
 */
export default function SimpleTable({head, body, minWidth, ...styleProps}: SimpleTableProps) {
  const table = (
    <Table
    {...styleProps}
      data={{head, body}}
      horizontalSpacing='sm'
      verticalSpacing='sm'
    />
  );

  if (minWidth) {
    return (
      <Table.ScrollContainer minWidth={minWidth} type='native'>
        {table}
      </Table.ScrollContainer>
    );
  }

  return table;
}